R/list_animal_ids_api.R

```diff
@@ -0,0 +1,27 @@
+#' List all available animal ids
+#'
+#' @param credentials Login credentials to the ETN database, as created by
+#'   `get_credentials()`
+#'
+#' @return A vector of all unique `id_pk` present in `common.animal_release`.
+#'
+#' @export
+list_animal_ids_api <- function(credentials = get_credentials()) {
+  api_domain <- "https://opencpu.lifewatch.be"
```

Comment:
I find the split between `api_domain` and `function_path` a bit arbitrary (parts are static in both) and would just combine them into one variable `endpoint` (I see you called it `api_url` in an example below.):
```R
endpoint <- "https://opencpu.lifewatch.be/library/etn/R/list_animal_ids"
```

Reply:
I'll combine them for now, but I might split them up later again. 

The code block where I create the response object is likely to be reused in other functions, thus might be wrapped in a helper with the body and function path (function name) as arguments. When I wrote this I was already experimenting with this approach, but decided not to go trough with it because it added unnecessary complexity at this stage (only one function actually using the API).

Reply:
Aha, I understand. A helper function would be good indeed.